packages/client/hooks/useGroupMatrix.ts

```diff
@@ -1,46 +1,46 @@
-import {SpotlightGroups_viewer} from './../__generated__/SpotlightGroups_viewer.graphql'
+import {SpotlightGroupsQueryResponse} from './../__generated__/SpotlightGroupsQuery.graphql'
 import {RefObject, useEffect, useState} from 'react'
 import {ElementWidth} from '../types/constEnums'
 import useResizeObserver from './useResizeObserver'
 import {MAX_SPOTLIGHT_COLUMNS} from '~/utils/constants'
 import useInitialRender from './useInitialRender'
 
-type Group = SpotlightGroups_viewer['similarReflectionGroups'][0]
+type Group = SpotlightGroupsQueryResponse['viewer']['similarReflectionGroups'][0]
 
 const useGroupMatrix = (
   resultsGroups: readonly Group[],
   resultsRef: RefObject<HTMLDivElement>,
   phaseRef: RefObject<HTMLDivElement>
 ) => {
   const isInit = useInitialRender()
+  const [groupMatrix, setGroupMatrix] = useState<Group[][]>([])
 
   const getColumnsCount = () => {
     const {current: el} = resultsRef
     const width = el?.clientWidth
     if (!width) return null
     let colCount = 1
+    const currentMaxColumns = Math.min(resultsGroups.length, MAX_SPOTLIGHT_COLUMNS)
     const getNextWidth = (count: number) =>
       ElementWidth.MEETING_CARD * count + ElementWidth.MEETING_CARD_MARGIN * (count - 1)
-    while (getNextWidth(colCount + 1) < width && colCount + 1 <= MAX_SPOTLIGHT_COLUMNS) {
+    while (getNextWidth(colCount + 1) < width && colCount + 1 <= currentMaxColumns) {
       colCount++
     }
     return colCount
   }
 
-  const initMatrix = (): Group[][] => {
+  const initMatrix = () => {
     const columnsCount = getColumnsCount()
-    if (columnsCount === null) return []
```

Comment:
When we update the search query, `queryRef` is briefly undefined which means we don't render the results. This means that the `resultsRef` is undefined so the groupMatrix is set to `[]`. To prevent this, I've moved `initMatrix` to the `useEffect` and only update state if `columnsCount` isn't null.